packages/workbox-background-sync/src/lib/QueueDb.ts

```diff
@@ -0,0 +1,167 @@
+/*
+  Copyright 2021 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {openDB, DBSchema, IDBPDatabase} from 'idb';
+import {RequestData} from './StorableRequest.js';
+import '../_version.js';
+
+interface QueueDBSchema extends DBSchema {
+  requests: {
+    key: number;
+    value: UnidentifiedQueueStoreEntry;
+    indexes: {queueName: string};
+  };
+}
+
+const DB_VERSION = 3;
+const DB_NAME = 'workbox-background-sync';
+const REQUEST_OBJECT_STORE_NAME = 'requests';
+const QUEUE_NAME_INDEX = 'queueName';
+
+export interface UnidentifiedQueueStoreEntry {
+  requestData: RequestData;
+  timestamp: number;
+  id?: number;
+  queueName?: string;
+  metadata?: object;
+}
+
+export interface QueueStoreEntry extends UnidentifiedQueueStoreEntry {
+  id: number;
+}
+
+/**
+ * A class to interact directly an IndexedDB created specifically to save and 
+ * retrieve QueueStoreEntries. This class encapsulates all the schema details
+ * to store the representation of a Queue.
+ *
+ * @private
+ */
+
+export class QueueDb {
+  private _db: IDBPDatabase<QueueDBSchema> | null = null;
+
+  /**
+   * Add QueueStoreEntry to underlying db.
+   * 
+   * @param {UnidentifiedQueueStoreEntry} entry
+   */
+  async addEntry(entry: UnidentifiedQueueStoreEntry) {
+    const db = await this.getDb();
+    await db.add(REQUEST_OBJECT_STORE_NAME, entry);
```

Comment:
Can this be switched to

```
await db.add(REQUEST_OBJECT_STORE_NAME, entry as QueueStoreEntry);
```

I think if we just "fake" the type info in that one place, and everywhere that we read from IDB we get back a `QueueStoreEntry`, that would be cleaner overall.